packages/@lwc/engine/src/framework/base-lightning-element.ts

```diff
@@ -196,44 +186,48 @@ function BaseLightningElementConstructor(this: LightningElement) {
             `Component creation requires a DOM element to be associated to ${vmBeingConstructed}.`
         );
     }
-    const vm = vmBeingConstructed as VM;
+
+    const vm = vmBeingConstructed;
     const {
         elm,
         mode,
         def: { ctor },
     } = vm;
+
     const component = this;
-    vm.component = component;
-    vm.tro = getTemplateReactiveObserver(vm);
-    vm.oar = create(null);
-    // interaction hooks
-    // We are intentionally hiding this argument from the formal API of LWCElement because
-    // we don't want folks to know about it just yet.
+    const cmpRoot = elm.attachShadow({
+        mode,
+        delegatesFocus: !!ctor.delegatesFocus,
+        '$$lwc-synthetic-mode$$': true,
+    } as any);
+
+    vm.component = this;
```

Comment:
for some reason I feel that the association between vm and component should happen before attachShadow, in case that is attempting to do anything with the host, probably ok though. just saying! the fact that vm.cmpRoot is undefined at that point, it probably a good indication that this line is also ok. I think I'm talking myself out of it now! jajaja, I withdraw the concern.